index.js

```diff
@@ -50,8 +53,23 @@ function checkFragments(ast) {
         fragmentSourceMap[fragmentName] = {};
         fragmentSourceMap[fragmentName][sourceKey] = true;
       }
+
+      if (!astFragmentMap[sourceKey]) {
+        astFragmentMap[sourceKey] = true;
+        return true;
+      } else {
+        // This fragmentDefinition already exists in the AST, no need to add it again.
+        return false;
+      }
+    } else {
+      // This definition is not a fragment.
+      return true;
     }
-  }
+  });
+
+  return assign({}, ast, {
+    definitions: uniqueFragmentDefinitions
```

Comment:
optionally, i can just call `lodash#uniq` on all the definitions instead of creaeting the map (in fact, ill probably do that if we're already importing lodash)